ios/packages/core/Sources/Types/Hooks/DataControllerHooks.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  DataControllerHooks.swift
+//  PlayerUI
+//
+//  Created by Zhao Xia Wu on 2024-04-08.
+//
+
+import Foundation
+import JavaScriptCore
+
+/**
+Hooks that can be tapped into for the DataController
+This lets users tap into events in the JS environment
+*/
+public struct DataControllerHooks {
+    /// Fired when the data changes
+    public var onUpdate: HookDecode<[Update]>
+}
+
+public struct Update: Decodable {
+    /** The updated binding */
+    var binding: BindingInstance
```

Comment:
can you add a regression test also to make sure that `binding` is decoded properly with a value?